ui/apps/platform/src/Containers/Clusters/Components/CredentialExpirationWidget.tsx

```diff
@@ -0,0 +1,52 @@
+import { Flex, FlexItem } from '@patternfly/react-core';
+import React from 'react';
+import { ClusterStatus } from '../clusterTypes';
+import CredentialExpiration from './CredentialExpiration';
+import CredentialInteraction from './CredentialInteraction';
+import ManageTokensButton from './ManageTokensButton';
+
+import { isCertificateExpiringSoon } from '../cluster.helpers';
+
+type CredentialExpirationWidgetProps = {
+    clusterId: string;
+    status: ClusterStatus;
+    isManagerTypeNonConfigurable: boolean;
+};
+
+const CredentialExpirationWidget = ({
+    clusterId,
+    status,
+    isManagerTypeNonConfigurable,
+}: CredentialExpirationWidgetProps) => {
+    const certExpiryStatus = status?.certExpiryStatus;
+    if (!certExpiryStatus?.sensorCertExpiry) {
+        return <CredentialExpiration certExpiryStatus={certExpiryStatus} />;
+    }
+
+    if (isManagerTypeNonConfigurable) {
+        const currentDatetime = new Date();
+        if (!isCertificateExpiringSoon(certExpiryStatus, currentDatetime)) {
+            return <CredentialExpiration certExpiryStatus={certExpiryStatus} />;
+        }
+        return (
+            <Flex direction={{ default: 'column' }}>
+                <FlexItem>
+                    <CredentialExpiration certExpiryStatus={certExpiryStatus} />
+                </FlexItem>
+                <FlexItem>
+                    <ManageTokensButton />
+                </FlexItem>
+            </Flex>
+        );
+    }
+
+    return (
+        <CredentialInteraction
+            certExpiryStatus={certExpiryStatus}
+            upgradeStatus={status?.upgradeStatus}
+            clusterId={clusterId}
+        />
+    );
```

Comment:
It would be nice to have semantic comments for each of these possible paths. (For instance, I am confused about why the first two paths appear to be return the exact same thing: `return <CredentialExpiration certExpiryStatus={certExpiryStatus} />;` )

Reply:
Added comments and simplified the return paths